doc/read-the-docs-site/doc/marconi-as-a-library/tutorials/BasicApp.hs

```diff
@@ -0,0 +1,496 @@
+  Core.createWorker
+    "BlockInfo" -- Name of the indexer. Useful for logging.
+    getEventsFromBlock -- Converting the block into the event
+    mkBlockInfoListIndexer
+
+-- BLOCKEND ListIndexer worker
+
+-- BLOCKSTART query
+
+-- The query parameter datatype.
+newtype GetBlockInfoFromBlockNoQuery = GetBlockInfoFromBlockNoQuery C.BlockNo
+  deriving stock (Eq, Ord, Show, Generic)
+
+-- We return the full 'BlockInfoEvent' as a response.
+-- The result is @Nothing@ if we provide a slot number which is in an unknown
+-- future, or if no block was created at the requested slot.
+type instance
+  Core.Result GetBlockInfoFromBlockNoQuery =
+    Maybe (Core.Timed C.ChainPoint BlockInfoEvent)
+
+-- BLOCKEND query
+
+-- BLOCKSTART ListIndexer query
+
+-- | Query the in-memory indexer
+instance
+  (Monad m)
+  => Core.Queryable
+      m
+      BlockInfoEvent -- The event type of the indexer
+      GetBlockInfoFromBlockNoQuery -- The query type
+      Core.ListIndexer -- The indexer backend
+  where
+  query
+    :: Core.Point BlockInfoEvent -- Give me what you know up to this point, potentially a point in future
+    -> GetBlockInfoFromBlockNoQuery -- The query we want to resolve
+    -> Core.ListIndexer BlockInfoEvent -- The backend
+    -> m (Core.Result GetBlockInfoFromBlockNoQuery)
+  -- There is not data at genesis. Return 'Nothing'.
+  query C.ChainPointAtGenesis _ _ = pure Nothing
+  query cp (GetBlockInfoFromBlockNoQuery bn) lsIndexer = do
+    -- Get all events from the ListIndexer which occurred *before* the
+    -- requested point @cp@.
+    let events = lsIndexer ^. Core.events
+        eventsBeforePoint = filter (\e -> e ^. Core.point <= cp) events
+
+    -- Then, we find the event which has the same block number as the one
+    -- requested.
+    pure $ List.find (\e -> blockNo (e ^. Core.event) == bn) eventsBeforePoint
+
+-- BLOCKEND ListIndexer query
+
+-- BLOCKSTART SQLiteIndexer
+
+-- | Creation of the worker which wraps the 'Core.SQLiteIndexer' of our indexer.
+mkBlockInfoSqliteIndexerWorker
+  :: FilePath
+  -- ^ Path of the SQLite database file
+  -> IO
+      ( Core.WorkerIndexer
+          IO
+          BlockEvent
+          -- \^ Input provided by the local node's chain-sync protocol
+          BlockInfoEvent
+          -- \^ Event type of the indexer
+          Core.SQLiteIndexer
+      )
+-- \^ Backend of the indexer
+
+mkBlockInfoSqliteIndexerWorker dbPath = do
+  ix <- Utils.toException $ mkBlockInfoSqliteIndexer dbPath
+  Core.createWorker "BlockInfo" getEventsFromBlock ix
+
+-- | Creation of 'Core.SQLiteIndexer' for the 'BlockInfoEvent' indexer.
+mkBlockInfoSqliteIndexer
+  :: (MonadIO m, MonadError Core.IndexerError m)
+  => FilePath
+  -- ^ Path of the SQLite database file
+  -> m (Core.SQLiteIndexer BlockInfoEvent)
+mkBlockInfoSqliteIndexer dbPath = do
+  -- This is actually a helper function on top of 'Core.mkSqliteIndexer' which
+  -- stores all indexed points from the chain. This is particularly useful when
```

Comment:
Not anymore: now it stores the last stable point and store it to resume from there.